Classes/EventListener/EventListenerLocator.php

```diff
@@ -136,17 +137,17 @@ protected static function detectListeners(ObjectManagerInterface $objectManager)
                 $parameters = array_values($reflectionService->getMethodParameters($listenerClassName, $listenerMethodName));
 
                 if (!isset($parameters[0])) {
-                    throw new Exception(sprintf('Invalid listener in %s::%s the method signature is wrong, must accept an EventInterface and optionally a EventMetadata', $listenerClassName, $listenerMethodName), 1472500228);
+                    throw new Exception(sprintf('Invalid listener in %s::%s the method signature is wrong, must accept an EventInterface and optionally a StoredEvent', $listenerClassName, $listenerMethodName), 1472500228);
                 }
                 $eventClassName = $parameters[0]['class'];
                 if (!$reflectionService->isClassImplementationOf($eventClassName, EventInterface::class)) {
                     throw new Exception(sprintf('Invalid listener in %s::%s the method signature is wrong, the first parameter should be cast to an implementation of EventInterface', $listenerClassName, $listenerMethodName), 1472504443);
                 }
 
                 if (isset($parameters[1])) {
-                    $metaDataType = $parameters[1]['class'];
-                    if ($metaDataType !== EventMetadata::class) {
-                        throw new Exception(sprintf('Invalid listener in %s::%s the method signature is wrong, the second parameter should be cast to EventMetadata', $listenerClassName, $listenerMethodName), 1472504303);
+                    $storedEventDataType = $parameters[1]['class'];
+                    if ($storedEventDataType !== StoredEvent::class) {
+                        throw new Exception(sprintf('Invalid listener in %s::%s the method signature is wrong, the second parameter should be cast to StoredEvent but expects an instance of "%s"', $listenerClassName, $listenerMethodName, $storedEventDataType), 1472504303);
```

Comment:
Mh, but it's actually "less" restored/hydrated than the converted event you get as first param.
Thinking about it again, I think it should be sth like `RawEvent` or `EventData` (which would also be more in sync if you use the framework without event-sourcing)